GetStream/stream-chat-react-native
# fix: delete message action for failed messages [CRNS-491]

## 🎯 Goal

Fixes: https://github.com/GetStream/stream-chat-react-native/issues/1020

## 🛠 Implementation details

### Fix

- Check if message is failed before calling `channel.deleteMessage()`
- If the message is failed, then simply remove it from the channel state instead of making an api call.

https://github.com/GetStream/stream-chat-react-native/pull/1048/files#diff-602d723a2e421edc15d642ead86f45ec181d85f5fbf575b0f94834d1f5c6cd58

### Refactoring

- Abstracted out all the message status types to separate const - `MessageStatusTypes`. This const can not be defined on js client, since status is technically a custom field and can be set by sdk with whatever value it prefers.

```tsx
const MessageStatusTypes = {
  FAILED: 'failed',
  RECEIVED: 'received',
  SENDING: 'sending',
};
```

## 🎨 UI Changes

https://user-images.githubusercontent.com/11586388/145457639-78079806-079e-4c2f-abfe-8d2987255318.mp4


## 🧪 Testing

- Open a channel
- Turn off your network
- Send a new message on channel, which will fail.
- Try to delete the message and notice that it doesn't get removed from the list (before this change)
